src/api/rivals.ts

```diff
@@ -0,0 +1,82 @@
+import apiClient from './apiClient';
+
+// 검색 응답 타입
+export interface RivalSearchResponse {
+  userId: number;
+  email: string;
+  name: string;
+}
+
+// 프로필 응답 타입
+export interface RivalProfileResponse {
+  userId: number;
+  email: string;
+  name: string;
+  intro?: string;
+  dailyQDays?: number;
+  answeredQuestions?: number;
+}
+
+// 팔로잉/팔로워 목록 아이템
+export interface RivalUserItem {
+  userId: number;
+  name: string;
+  email: string;
+}
+
+// 팔로잉/팔로워 목록 응답
+export interface RivalListResponse {
+  items: RivalUserItem[];
+  nextCursor: number;
+  hasNext: boolean;
+}
+
+// 라이벌 검색
+export const searchRival = async (email: string): Promise<RivalSearchResponse> => {
+  const response = await apiClient.get<RivalSearchResponse>('/api/rivals/search', {
+    params: { email }
+  });
+  return response.data;
+};
+
+// 라이벌 프로필 조회
+export const getRivalProfile = async (userId: number): Promise<RivalProfileResponse> => {
+  const response = await apiClient.get<RivalProfileResponse>(`/api/rivals/${userId}/profile`);
+  return response.data;
+};
+
+// 팔로잉 목록 조회
+export const getFollowingList = async (lastId?: number, limit: number = 10): Promise<RivalListResponse> => {

```

Comment:
옵셔널 매개변수가 여러 개일 경우 특정 매개변수만 전달하려면 이전 인자 자리에 `undefined`를 명시해야 하는 불편함이 있습니다.
이런 경우에는 매개변수를 객체 형태로 받는 방식이 직관적이고 확장에도 유리해요.

```suggestion
export const getFollowingList = async (params: {
  lastId?: number,
  limit?: number
}): Promise<RivalListResponse> => {
  const { lastId, limit = 10 } = params;
```